ipatests/test_integration/test_subids.py

```diff
@@ -268,3 +269,30 @@ def test_sudid_match_shows_uid(self):
                                           f"--subuid={subuid}"])
         owner = self._parse_result(result)["owner"]
         assert owner == uid
+
+    def test_nsswitch_doesnot_contain_subid_entry(self):
+        """
+        This testcase checks that when ipa-client-install
+        is installed without subid option, the nsswitch.conf
+        doesnot contain subid entry
+        """
+        cmd = self.clients[0].run_command(
+            ["grep", "^subid", "/etc/nsswitch.conf"]
+        )
+        assert cmd.returncode == 1
+        assert 'subid' not in cmd.stdout_text
+
+    def test_nsswitch_is_updated_with_subid_entry(self):
+        """
+        This test case checks that when ipa-client-install
+        is installed with --subid option, the nsswitch.conf
+        file is modified with the entry 'subid: sss'
+        """
+        tasks.uninstall_client(self.clients[0])
+        tasks.install_client(self.master, self.clients[0],
+                             extra_args="--subid")
```

Comment:
`extra_args` expects an iterable (usually a list). As a string value is iterable (it returns one char at a time), it will be treated as `ipa-client-install - - s u b i d`. Replace with
`extra_args=['--subid']`